### PR TITLE
kubectl add show-labels flag to make it more readable

### PR DESCRIPTION
**Pull request by adohe-zz**
FYI. reference this #19217 for more detail.

### CODE REVIEW

**Comment by AdoHe** [comment]
I I signed it!

**Comment by AdoHe** [comment]
From the travis-ci, it seems the make process failed. Is is related to how I open a PR?

**Comment by bgrant0607** [comment]
@AdoHe Please run hack/update-generated-docs.sh

The kubectl docs are generated from help text.

**Comment by bgrant0607** [comment]
cc @kubernetes/kubectl

**Comment by bgrant0607** [comment]
BTW, you can see the instructions at the bottom of the page linked from the travis status:
https://travis-ci.org/kubernetes/kubernetes/builds/102558621

**Comment by bgrant0607** [comment]
@andronat Do you have time to review this?

**Comment by andronat** [comment]
@bgrant0607 I can take a look but whom do I ping for adding the labels?

**Comment by bgrant0607** [comment]
@andronat Either @janetkuo or I can add the labels.

**Comment by AdoHe** [comment]
@bgrant0607 Ok, I will do this.

**Comment by kargakis** [comment]
@kubernetes/kubectl

**Comment by AdoHe** [comment]
@andronat guess I have to run hack/update-generated-docs.sh first, I will do it but I am wondering whether I have to recreate a pull request, since I have to update some docs and make a new commit? do you have any good idea about this?

**Comment by andronat** [comment]
Hello @AdoHe, you don't need to make a new pull request but only to update your fork. To do that firstly finish your commits, commit them locally and then use `git push --force`. Google it a little bit before though so you make sure that you understand what is doing exactly.

Secondly please read this [link](https://github.com/kubernetes/kubernetes/blob/4ca66d2aefa20c27b670b2fa890052daadc05294/docs/devel/development.md#committing-changes-to-your-fork) and add the provided hooks into your git workflow as it runs all the tests you need to pass.

File: pkg/kubectl/resource_printer.go [hunk 1]
```diff
@@ -1540,6 +1617,16 @@ func formatWideHeaders(wide bool, t reflect.Type) []string {
 	return nil
 }
 
+// headers for --show-labels=true
+func formatAllLablesHeader(showLabels bool, t reflect.Type) []string {
```
Comment (kargakis) [inline pkg/kubectl/resource_printer.go]: typo

Comment (AdoHe) [inline pkg/kubectl/resource_printer.go]: @kargakis how about change this to `formatShowLabelsHeader`?

Comment (kargakis) [inline pkg/kubectl/resource_printer.go]: SGTM

**Comment by kargakis** [comment]
We already have `-o wide` for printing the node name for pods. @bgrant0607 if we are going to put more info under flags in the future (most likely) how about a more generic flag name like `--verbose`?

**Comment by AdoHe** [comment]
@kargakis we add a flag here just to control the labels display to make the resource result more readable, of course s more generic flag name is considerable.

**Comment by AdoHe** [comment]
@andronat seems there are some conflicts since I use `git push --force`, the docs should be updates and all the pre-commit hooks are ok. Guess we have to fix the conflict.

**Comment by andronat** [comment]
Yes, you need to rebase your branch. Rebase your current working branch from the project's master (be careful not to be confused with your fork's master). You can find some links [here](https://robots.thoughtbot.com/keeping-a-github-fork-updated), [here](https://help.github.com/articles/syncing-a-fork/) and [here](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request).

**Comment by bgrant0607** [comment]
@kargakis I intentionally didn't want `--show-labels` under a generic flag like `-o wide` or `--verbose` because I wanted users to be able to choose between showing all the labels or individual label values via `-L` independent of any other information.

**Comment by AdoHe** [comment]
@andronat I just rebase my branch, the conflicts get resolved, would you mind reviewing this?

**Comment by AdoHe** [comment]
@andronat I just update this according to your suggestions.

**Comment by andronat** [comment]
Ok cool just fix this last typo and squeeze all the commits in one. If all tests pass we can merge.

**Comment by andronat** [comment]
Hi @janetkuo do you know why the Submit Queue is always pending? Is this normal? The PR LGTM but we need to see all tests pass right?

**Comment by smarterclayton** [comment]
ok to test

**Comment by erictune** [comment]
@andronat the submit queue was blocked due to some failing tests at head.  It has been fixed this morning.

@adohe you need to rebase.   See: 
https://github.com/kubernetes/kubernetes/blob/4ca66d2aefa20c27b670b2fa890052daadc05294/docs/devel/development.md#keeping-your-development-fork-in-sync

**Comment by AdoHe** [comment]
@erictune ok, I will do rebase later.

**Comment by bgrant0607** [comment]
```
_output/local/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/cmd_test.go:318: not enough arguments in call to kubectl.NewHumanReadablePrinter
```

**Comment by AdoHe** [comment]
the Jenkins failed, I can just find the type checker error [here](https://00e9e64bac29de6c8dca35fce8536ded7e29240d2ab5411eda-apidata.googleusercontent.com/download/storage/v1_internal/b/kubernetes-jenkins/o/pr-logs%2Fc165cd64569339d93eef483f2beef2e4db35a14d%2Fkubernetes-pull-test-unit-integration%2F10932%2Fbuild-log.txt?qk=AD5uMEv-MZ4sBkb7QHRb-5QVmLbmw6Er0ziONIISfqEnUqKsZHUUAYk-O44nT25f68hXy2qlnkztjoMoVeVe6VPLt_8vU3CEGU2oYLO9Cua5X7u6kIFsLDPr7ysihVQPBmgRUlD8RZBsr27pPlk0cKx-yom_3wI11R4PEQ0VLlBcyfba7exFzqp-xtOwI5HVf0y-FeaVH_HNjz90sLwmSqh-Qvyc_9GyYYBBHg3QKXWCGUha3kmVgPMOmjFb2JQVL1e6l9p2DD5IARSCSPzsKJNMMQZZb2K5zuK3qCohyFs4WstaF5Y2pz9bfWqCLLhoDUTAZJp_H_KX1WtE8ix7c5dBtarY0qtELha-CzKtUQDl4PX6cQ4mippXg91yzYMeEQgrDwfIJj-oL2e3mWLBgVM8F4i51_N0lTsiwnYxmqdM2hfd9Gkraau2nEU2kTL1TfL5J2cpbhil2jk5u9yO9Wj-wqj2gPNIy83ojIdpe4_eCtjR2VTXJ5lW24fgp0DkD6MFf99OdNe13sIVW7gqjYOGoCbakwtWeciFhwlCXJ4-aScwe2TgFEfDXD8nrkwAXiWVIeqIz-i_7F4AO7OQDvL1l5DTYco3wiInIt9TyJagXwd9dQ-LDjHFEhFHcCymktR-YwCNzKRYPR7pzXAvAsCBNPohgZXRHk4YcOcIKUPrmDbSC4aqBmPW6oUoEqKkholmecByPVRP2Wes1miVlNkJQ8aCURig7xmG_d90E7So4T2hSAdYSkd4Va_ma7KwlQKjgOWxGdvgLB5FTqsQxYou6QhCG2L93T1-X2C0D_eDhIdgGeF_PRW8rE5ShK2sYsjDExy6bQmhf65bQkUTpQ4K8-7XH4TJ8qbkI0n2nimNRHVa966jgEUWG-kbgOHcQm4IFx7se39MglhxJ0b1QzAxrKV7upEswPgkjs1oJ00FY0ZNw6hQPNoznRIqquI-Hz7fytKuk1mws9PpCB9eCXb0od4K5mYIsQ).

**Comment by andronat** [comment]
Did you try to run the tests locally? `./hack/test-go.sh` seems to fail. If all tests pass locally to your computer, rebase and force push again so Jenkins can reboot.

**Comment by AdoHe** [comment]
I will fix this today.

**Comment by andronat** [comment]
lgtm @janetkuo

**Comment by andronat** [comment]
ok to test

**Comment by AdoHe** [comment]
rebase again? I have did rebase, why still needs rebase?

**Comment by bgrant0607** [comment]
@AdoHe Sorry about the rebases. It's because we check in the generated documentation. We'd like to fix that, but haven't had time. Right now, the submit queue is stalled due to e2e test failures. Once this is likely to merge soon I'll ask you to rebase and regenerate docs again.

**Comment by AdoHe** [comment]
@bgrant0607 Ok, no problem I will do that later.

**Comment by bgrant0607** [comment]
Tests are passing and submit queue is working. Please rebase now @AdoHe.

**Comment by AdoHe** [comment]
@bgrant0607 Just rebase this PR.

**Comment by bgrant0607** [comment]
LGTM

**Comment by bgrant0607** [comment]
@k8s-oncall If the queue doesn't merge this soon, could we please get a manual merge? We have several other PRs changing kubectl output that will need to be rebased.

**Comment by saad-ali** [comment]
@k8s-bot ok to test

**Pull request merged**